client/src/main/scala/com/dwolla/cloudflare/domain/model/ratelimits/RateLimitAction.scala

```diff
@@ -1,10 +1,26 @@
 package com.dwolla.cloudflare.domain.model.ratelimits
 
-case class RateLimitAction (
-  mode: String,
-  timeout: Int,
-  response: Option[RateLimitActionResponse] = None
-)
+sealed trait RateLimitAction {
+  val mode: String
+  val timeout: Option[Int]
+  val response: Option[RateLimitActionResponse]
+}
+
+case class TimeoutRateLimitAction(mode: String, timeoutSeconds: Int, response: Option[RateLimitActionResponse] = None) extends RateLimitAction {
+  override val timeout: Option[Int] = Some(timeoutSeconds)
+}
+
+case class ChallengeRateLimitAction() extends RateLimitAction {
+  override val mode = "challenge"
+  override val timeout: Option[Int] = None
```

Comment:
Why do we want a `timeout` field to be present here at all? It seems like kind of a non sequitur—to me an timeout set to `None` implies an infinite timeout, instead of a place where the idea of a timeout doesn't apply? Maybe I'm missing something.

Reply:
Symptom of the DTO not matching the ADT.